tests/unit/lib_test.php

```diff
@@ -269,4 +269,99 @@ public function test_turnitintooltwo_get_report_gen_speed_params() {
 
 	    $this->assertEquals($expected, $result);
     }
+
+    public function test_turnitintooltwo_availability_status() {
+        global $DB;
+
+        $this->resetAfterTest();
+
+        $this->setAdminUser();
+
+        // Create a course and assignment.
+        $course = $this->getDataGenerator()->create_course();
+
+        $turnitintooltwoassignment = $this->make_test_tii_assignment();
+        $cmid = $this->make_test_module($turnitintooltwoassignment->turnitintooltwo->course,'turnitintooltwo', $turnitintooltwoassignment->turnitintooltwo->id);
+        $context = context_module::instance($cmid);
+        $cm = $DB->get_record("course_modules", array('id' => $cmid));
+
+        // Set to the student user.
+        $generator = $this->getDataGenerator();
+
+        // Create users.
+        $student = self::getDataGenerator()->create_user();
+        $teacher = self::getDataGenerator()->create_user();
+
+        // Users enrolments.
+        $studentrole = $DB->get_record('role', array('shortname' => 'student'));
+        $teacherrole = $DB->get_record('role', array('shortname' => 'editingteacher'));
+        $this->getDataGenerator()->enrol_user($student->id, $course->id, $studentrole->id, 'manual');
+        $this->getDataGenerator()->enrol_user($teacher->id, $course->id, $teacherrole->id, 'manual');
+
+        self::setUser($student);
+
+        // Create data object for passing in.
+        $data = new stdClass();
+        $data->id = $cm->instance;
+        $data->timeclose = 0;
+        $data->timeopen = 0;
+
+        // Usual case.
+        list($status, $warnings) = mod_turnitintooltwo_get_availability_status($data, false);
+        $this->assertEquals(true, $status);
+        $this->assertCount(0, $warnings);
+
+        // Turnitintooltwo not open.
+        $data->timeopen = time() + DAYSECS;
```

Comment:
if you make the change recommended above, this test will break as well. I do not think it is a good test to check on the number of elements in the `$warnings` associative array. Look for the specific key, which you know should exists only in the event the specific error occurs.

Something like `$this->assertTrue(array_key_exists('notopenyet', $warnings));`